collector/collector.go

```diff
@@ -395,6 +406,30 @@ func (nc *NATSCollector) objectToMetrics(response map[string]interface{}, namesp
 	}
 }
 
+// mapKeys returns a map of all keys in a map, including nested maps.
+// The keys from nested maps are prefixed with the parent key.
+func mapKeys(input map[string]interface{}, prefix string) map[string]struct{} {
+	keys := make(map[string]struct{})
+
+	for k, v := range input {
+		fullKey := k
+		if prefix != "" {
+			fullKey = prefix + "_" + k
+		}
+
+		if nestedMap, ok := v.(map[string]interface{}); ok {
+			nestedKeys := mapKeys(nestedMap, fullKey)
+			for nestedKey := range nestedKeys {
+				keys[nestedKey] = struct{}{}
```

Comment:
we dont't have to have the same prefix logic for nested maps?

Reply:
The prefix logic works for keys in nested maps (`nestedKeys`). We don't add the map key itself to the result as it will not be used as a metric. I've added a test for this function which should showcase what's expected:
```go
func TestMapKeys(t *testing.T) {
	m := map[string]any{
		"foo": "bar",
		"baz": "quux",
		"nested": map[string]any{
			"foo": "bar",
			"baz": "quux",
			"nested": map[string]any{
				"foo": "bar",
				"baz": "quux",
			},
		},
	}
	expected := map[string]struct{}{
		"foo":               {},
		"baz":               {},
		"nested_foo":        {},
		"nested_baz":        {},
		"nested_nested_foo": {},
		"nested_nested_baz": {},
	}
	keys := mapKeys(m, "")
	if !maps.Equal(keys, expected) {
		t.Fatalf("expected %v, got %v", expected, keys)
	}
}
```